src/main/java/school/hei/haapi/endpoint/rest/controller/TeacherController.java

```diff
@@ -40,19 +42,19 @@ public Teacher getTeacherById(
   @GetMapping(value = "/teachers")
   public List<Teacher> getTeachers(
       @RequestParam PageFromOne page, @RequestParam("page_size") BoundedPageSize pageSize) {
-    return userService
-        .getByRole(User.Role.TEACHER, page, pageSize).stream()
+    return userService.getByRole(User.Role.TEACHER, page, pageSize).stream()
         .map(userMapper::toRestTeacher)
         .collect(toUnmodifiableList());
   }
 
   @PutMapping(value = "/teachers")
   public List<Teacher> createOrUpdateTeachers(@RequestBody List<Teacher> toWrite) {
-    var saved = userService.saveAll(toWrite.stream()
-        .map(userMapper::toDomain)
-        .collect(toUnmodifiableList()));
-    return saved.stream()
-        .map(userMapper::toRestTeacher)
-        .collect(toUnmodifiableList());
+    if (toWrite.size() > EventConf.MAX_EVENT_REQUEST_ENTRY) {
+      throw new BadRequestException("teachers to create or update must be ≤ 10");
+    }
+    var saved =
```

Comment:
Do not modify indentation unless it's really really bad. It just makes reviewing the PR harder

Reply:
still google_check format